satellite/payments/stripecoinpayments/service.go

```diff
@@ -126,8 +130,99 @@ func (service *Service) updateTransactions(ctx context.Context, ids coinpayments
 			},
 		)
 
-		// TODO: update stripe customer balance
+		// moment of transition to received state, which indicates
+		// that customer funds were accepted, so we can apply this
+		// amount to customer balance. So we create intent to update
+		// customer balance in the future.
+		if info.Status == coinpayments.StatusReceived {
+			applies = append(applies, id)
+		}
 	}
 
-	return service.transactionsDB.Update(ctx, updates)
+	return service.transactionsDB.Update(ctx, updates, applies)
+}
+
+// applyAccountBalanceLoop fetches all unapplied transaction in a loop, applying transaction
+// received amount to stripe customer balance.
+func (service *Service) updateAccountBalanceLoop(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	const (
+		limit = 25
+	)
+
+	before := time.Now()
+
+	txsPage, err := service.transactionsDB.ListUnapplied(ctx, 0, limit, before)
+	if err != nil {
+		return err
+	}
+
+	for _, tx := range txsPage.Transactions {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		if err = service.applyTransactionBalance(ctx, tx); err != nil {
+			return err
+		}
+	}
+
+	for txsPage.Next {
+		select {
```

Comment:
These can simply check for `if err := ctx.Err(); err != nil { return err }`

Reply:
Also is it safe to abort these in the middle of applying the transactions?

Reply:
The Idea is that consumed transactions won't appear on new cycle iteration